### PR TITLE
 sql: update tidb slow log operation docs

### PR DESCRIPTION
**Pull request by crazycs520**
This PR update TiDB slow log operation docs.

Cause TiDB after PR https://github.com/pingcap/tidb/pull/9290, change the slow log format and change slow-log output to a file separately by default.

### CODE REVIEW

**Comment by crazycs520** [comment]
@winkyao PTAL

File: sql/slow-query.md [hunk 1]
```diff
@@ -9,81 +9,136 @@ category: user guide
...
+* `Wait_time`：表示这个语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队，当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
+* `Backoff_time`：表示语句遇到需要重试的错误时在重试前等待的时间，常见的需要重试的错误有以下几种：遇到了 lock、Region 分裂、`tikv server is busy`。
+* `Request_count`：表示这个语句发送的 Coprocessor 请求的数量。
+* `Total_keys`：表示 Coprocessor 扫过的 key 的数量。
+* `Process_keys`：表示 Coprocessor 处理的 key 的数量。相比 total_keys，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
+* `DB`：表示当前的 database。
+* `Index_ids` ：表示语句涉及到的索引的 ID
+* `Is_internal`：表示是否是 TiDB 内部 SQL。true 为TiDB 内部执行的SQL, 比如 analyze，load variable 等；false 为用户执行的 SQL 。
+* `Digest`： 表示 SQL 语句的指纹
+* `Query`：表示 SQL 语句。
```
Comment (winkyao) [inline sql/slow-query.md]: 没有 `Query` 这个字段，或者你解释一下这个会出现在 `INFORMATION_SCHEMA.SLOW_QUERY`

Comment (crazycs520) [reply sql/slow-query.md]: done.

**Comment by crazycs520** [comment]
@qw4990 PTAL

File: sql/slow-query.md [hunk 2]
```diff
@@ -9,81 +9,194 @@ category: user guide
...
+# Time: 2019-03-18-12:10:19.513961 +0800
+# Txn_start_ts: 407078752230047745
+# User: root@127.0.0.1
+# Conn_ID: 1
+# Query_time: 16.479155653
+# Process_time: 5.634 Wait_time: 0.002 Request_count: 2 Total_keys: 20002 Process_keys: 20000
+# DB: test
+# Index_ids: [1]
+# Is_internal: false
+# Digest: 3635413fe0c8e1aa8307f4f018fe1a9325ea0b97452500106d3f6783fcb65e33
```
Comment (qw4990) [inline sql/slow-query.md]: Add
```
# Cop_process: Avg_time: 1s P90_time: 2s Max_time: 3s Max_addr: 10.6.131.78
# Cop_wait: Avg_time: 10ms P90_time: 20ms Max_time: 30ms Max_Addr: 10.6.131.79
# Memory_max: 4096
```
below `Digest`.

File: sql/slow-query.md [hunk 3]
```diff
@@ -9,81 +9,194 @@ category: user guide
...
+* `Process_time`：执行 SQL  在 TiKV 的处理时间之和，因为数据会并行的发到 TiKV 执行，这个值可能会超过 `Query_time`。
+* `Wait_time`：表示这个语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队，当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
+* `Backoff_time`：表示语句遇到需要重试的错误时在重试前等待的时间，常见的需要重试的错误有以下几种：遇到了 lock、Region 分裂、`tikv server is busy`。
+* `Request_count`：表示这个语句发送的 Coprocessor 请求的数量。
+* `Total_keys`：表示 Coprocessor 扫过的 key 的数量。
+* `Process_keys`：表示 Coprocessor 处理的 key 的数量。相比 total_keys，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
+* `DB`：表示当前的 database。
+* `Index_ids` ：表示语句涉及到的索引的 ID
+* `Is_internal`：表示是否是 TiDB 内部 SQL。true 为TiDB 内部执行的SQL, 比如 analyze，load variable 等；false 为用户执行的 SQL 。
+* `Digest`： 表示 SQL 语句的指纹
```
Comment (qw4990) [inline sql/slow-query.md]: `Memory_max`：表示执行期间做多时候使用的内存数量, 单位为byte.
`Cop_process `：表示cop-tasks的执行时间, 一共包含`Avg_time`, `P90_time`, `Max_time`, `Max_addr`几个子字段.
`Cop_wait`: 表示cop-tasks的请求等待时间, 一共包含`Avg_time`, `P90_time`, `Max_time`, `Max_addr`几个子字段.

麻烦添加在`Digest`下面..
另外`Digest`冒号后面多了一个空格... `User`和`Digest`最后没有句号. 😆

**Comment by lilin90** [comment]
@crazycs520 Please address comments. Thanks!

File: op-guide/tidb-config-file.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 ### `slow-query-file`
```
Comment (superlzs0476) [inline op-guide/tidb-config-file.md]: we should add an introduction about slow query log differents feature with master or before 2.1.8 on the official document.

File: sql/slow-query.md (hunk 3 above)
Comment (crazycs520) [reply sql/slow-query.md]: done.

File: sql/slow-query.md (hunk 2 above)
Comment (crazycs520) [reply sql/slow-query.md]: done.

File: op-guide/tidb-config-file.md (hunk 4 above)
Comment (winkyao) [reply op-guide/tidb-config-file.md]: The user can look into the docs of 2.1, I think no need to mention here.

File: sql/slow-query.md [hunk 5]
```diff
@@ -9,81 +9,202 @@ category: user guide
...
+目前查询 `INFORMATION_SCHEMA.SLOW_QUERY`  只会解析配置文件中 `slow-query-file` 设置的慢日志文件名，默认是 "tidb-slow.log" 。 但如果想要解析其他的日志文件，可以通过设置 session 变量 `tidb_slow_query_file` 为具体的文件路径，然后查询 `INFORMATION_SCHEMA.SLOW_QUERY` 就会按照设置的路径去解析慢日志文件。
+```sql
+/* 设置慢日志文件路径，方便解析其他的慢日志文件，tidb_slow_query_file 变量的作用域是 session */
+mysql test@127.0.0.1:test> set tidb_slow_query_file="/path-to-log/tidb-slow.log"
+Query OK, 0 rows affected
+Time: 0.001s
+```
 
-表示 SQL 语句
+#### 不足
```
Comment (winkyao) [inline sql/slow-query.md]: 不足跟 #### 解析其他的 TiDB 慢日志文件 可以合成一个，

File: op-guide/tidb-config-file.md (hunk 4 above)
Comment (crazycs520) [reply op-guide/tidb-config-file.md]: done.

File: sql/slow-query.md (hunk 5 above)
Comment (crazycs520) [reply sql/slow-query.md]: done.

File: sql/slow-query.md [hunk 6]
```diff
@@ -5,85 +5,206 @@ category: user guide
...
+* `Request_count`：表示这个语句发送的 Coprocessor 请求的数量。
+* `Total_keys`：表示 Coprocessor 扫过的 key 的数量。
+* `Process_keys`：表示 Coprocessor 处理的 key 的数量。相比 total_keys，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
+* `DB`：表示当前的 database。
+* `Index_ids` ：表示语句涉及到的索引的 ID
+* `Is_internal`：表示是否是 TiDB 内部 SQL。true 为TiDB 内部执行的SQL, 比如 analyze，load variable 等；false 为用户执行的 SQL 。
+* `Digest`：表示 SQL 语句的指纹。
+* `Memory_max`：表示执行期间做多时候使用的内存数量, 单位为byte。
+* `Num_cop_tasks`：表示cop-tasks的数目。
+* `Cop_process `：表示cop-tasks的执行时间, 一共包含`Avg_time`, `P90_time`, `Max_time`, `Max_addr`几个子字段。
```
Comment (qw4990) [inline sql/slow-query.md]: 昨天下午和 @crazycs520 又讨论了下, 目前这种格式不能让` pt-query`很好的解析, 于是又调整了下.
目前的格式如下:
```
# Cop_proc_avg: 1 Cop_proc_p90: 2 Cop_proc_max: 3 Cop_proc_addr: 10.6.131.78
```

辛苦再修改一下吧, 改成:
`Cop_proc_avg`: cop-task的平均执行时间.
`Cop_proc_p90`: cop-task的P90分位执行时间.
`Cop_proc_max`: cop-task的最大执行时间.
`Cop_proc_addr`: 执行时间最长的cop-task所在地址.
`Cop_wait_avg`: cop-task的平均等待时间.
`Cop_wait_p90`: cop-task的P90分位等待时间.
`Cop_wait_max`: cop-task的最大等待时间.
`Cop_wait_addr`: 等待时间最长的cop-task所在地址.

Comment (crazycs520) [reply sql/slow-query.md]: done.

File: op-guide/tidb-config-file.md [hunk 7]
```diff
@@ -57,7 +57,7 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 
 + 慢查询日志的文件名。
-+ 默认: ""
++ 默认: "tidb-slow.log" ，注：由于 TiDB V2.1.8 更新了慢日志格式，所以将慢日志单独输出到了慢日志文件。 V2.1.8 之前的版本，该变量的默认值是 ""。
```
Comment (bb7133) [inline op-guide/tidb-config-file.md]:
```suggestion
+ 默认: "tidb-slow.log"，注：由于 TiDB V2.1.8 更新了慢日志格式，所以将慢日志单独输出到了慢日志文件。V2.1.8 之前的版本，该变量的默认值是 ""。
```

File: sql/slow-query.md [hunk 8]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+TiDB 在 V2.1.8 之后更改了慢日志格式，V2.1.8 之前的版本请看[这个文档](<https://github.com/pingcap/docs-cn/blob/master/v2.1/sql/slow-query.md>)。
+
 一条不合理的 SQL 语句会导致整个集群压力增大，响应变慢。对于这种问题，需要用慢查询日志来定位有问题的语句，解决性能问题。
 
 ### 获取日志
 
-通过在 TiDB 的日志文件上 grep `SLOW_QUERY` 这个关键字，可以得到执行时间超过 [slow-threshold](../op-guide/tidb-config-file.md#slow-threshold) 的语句日志。
-
-`slow-threshold` 可以通过配置文件修改，默认是 300ms。如果配置了 [slow-query-file](../op-guide/tidb-config-file.md#slow-query-file)，慢查询日志会全部写在这个文件里。
+TiDB 会将执行时间超过 [slow-threshold](../op-guide/tidb-config-file.md#slow-threshold) 的语句默认单独输出到 [slow-query-file](../op-guide/tidb-config-file.md#slow-query-file) 文件中 ，并对慢日志的格式做了兼容，可以用 `pt-query-digest` 直接分析慢日志 文件。`slow-threshold` 可以通过配置文件修改，默认是 300ms。`slow-query-file` 默认是 `tidb-slow.log` 。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
TiDB 会将执行时间超过 [slow-threshold](../op-guide/tidb-config-file.md#slow-threshold) 的语句默认单独输出到 [slow-query-file](../op-guide/tidb-config-file.md#slow-query-file) 文件中，并对慢日志的格式做了兼容，可以用 `pt-query-digest` 直接分析慢日志文件。`slow-threshold` 可以通过配置文件修改，默认是 300ms。`slow-query-file` 默认是 `tidb-slow.log`。
```

File: sql/slow-query.md [hunk 9]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
-
-#### total_keys
-
-表示 Coprocessor 扫过的 key 的数量
-
-#### processed_keys
+* `Time`：表示日志打印时间。
+* `Txn_start_ts`：表示事务的开始时间戳，也是事务的 ID, 可以用这个值在日志中 grep 出事务相关的日志。
+* `User`：表示执行语句的用户名。
+* `Conn_ID`：表示 connection ID，即 session ID, 可以用类似 `con:3 ` 的关键字在  TiDB 日志中 grep 出 session ID 为 3 的日志。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
* `Conn_ID`：表示 connection ID，即 session ID, 可以用类似 `con:3` 的关键字在 TiDB 日志中 grep 出 session ID 为 3 的日志。
```

File: sql/slow-query.md [hunk 10]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Query_time`：表示执行这个语句花费的时间。只有执行时间超过 slow-threshold 的语句才会输出这个日志，单位是秒，一下所有的时间字段的单位都是秒。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
* `Query_time`：表示执行这个语句花费的时间。只有执行时间超过 slow-threshold 的语句才会输出这个日志，单位是秒，以下所有的时间字段的单位都是秒。
```

File: sql/slow-query.md [hunk 11]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Process_time`：执行 SQL  在 TiKV 的处理时间之和，因为数据会并行的发到 TiKV 执行，这个值可能会超过 `Query_time`。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
* `Process_time`：执行 SQL 在 TiKV 的处理时间之和，因为数据会并行的发到 TiKV 执行，这个值可能会超过 `Query_time`。
```

File: sql/slow-query.md [hunk 12]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Wait_time`：表示这个语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队，当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
* `Wait_time`：表示这个语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队；当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
```

File: sql/slow-query.md [hunk 13]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Backoff_time`：表示语句遇到需要重试的错误时在重试前等待的时间，常见的需要重试的错误有以下几种：遇到了 lock、Region 分裂、`tikv server is busy`。
+* `Request_count`：表示这个语句发送的 Coprocessor 请求的数量。
+* `Total_keys`：表示 Coprocessor 扫过的 key 的数量。
+* `Process_keys`：表示 Coprocessor 处理的 key 的数量。相比 total_keys，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
+* `DB`：表示当前的 database。
+* `Index_ids` ：表示语句涉及到的索引的 ID
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
* `Index_ids`：表示语句涉及到的索引的 ID。
```

File: sql/slow-query.md [hunk 14]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Is_internal`：表示是否是 TiDB 内部 SQL。true 为TiDB 内部执行的SQL, 比如 analyze，load variable 等；false 为用户执行的 SQL 。
+* `Digest`：表示 SQL 语句的指纹。
+* `Memory_max`：表示执行期间做多时候使用的内存数量, 单位为byte。
+* `Num_cop_tasks`：表示cop-tasks的数目。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
* `Num_cop_tasks`：表示 cop-tasks 的数目。
```

File: sql/slow-query.md [hunk 15]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Cop_proc_avg`: cop-task的平均执行时间。
+* `Cop_proc_p90`: cop-task的P90分位执行时间。
+* `Cop_proc_max`: cop-task的最大执行时间。
+* `Cop_proc_addr`: 执行时间最长的cop-task所在地址。
+* `Cop_wait_avg`: cop-task的平均等待时间。
+* `Cop_wait_p90`: cop-task的P90分位等待时间。
+* `Cop_wait_max`: cop-task的最大等待时间。
+* `Cop_wait_addr`: 等待时间最长的cop-task所在地址。
```
Comment (bb7133) [inline sql/slow-query.md]: Please add necessary whitespaces between English words and Chinese characters for all sentences above

File: sql/slow-query.md [hunk 16]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+* `Query`：表示 SQL 语句。慢日志里面不会打印 `Query`，但映射到内存表后，对应的字段叫 `Query`。
+
+### 慢日志内存映射表 
+为了方便用 SQL 查询定位慢查询，TiDB 将慢日志内容解析后映射到 `INFORMATION_SCHEMA.SLOW_QUERY` 表中，表中 column 名和慢日志中记录的字段名一一对应 。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
为了方便用 SQL 查询定位慢查询，TiDB 将慢日志内容解析后映射到 `INFORMATION_SCHEMA.SLOW_QUERY` 表中，表中 column 名和慢日志中记录的字段名一一对应。
```

File: sql/slow-query.md [hunk 17]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+| select * from t1 where a>3; | 0.50077719  | t1:pseudo                       |
+| select * from t1 join t2;   | 0.931260518 | t1:407872303825682445,t2:pseudo |
++-----------------------------+-------------+---------------------------------+
+```
 
-表示语句涉及到的索引的 ID
+#### 解析其他的 TiDB 慢日志文件
 
-#### sql
+目前查询 `INFORMATION_SCHEMA.SLOW_QUERY`  只会解析配置文件中 `slow-query-file` 设置的慢日志文件名，默认是 "tidb-slow.log" 。 但如果想要解析其他的日志文件，可以通过设置 session 变量 `tidb_slow_query_file` 为具体的文件路径，然后查询 `INFORMATION_SCHEMA.SLOW_QUERY` 就会按照设置的路径去解析慢日志文件。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
目前查询 `INFORMATION_SCHEMA.SLOW_QUERY` 只会解析配置文件中 `slow-query-file` 设置的慢日志文件名，默认是 "tidb-slow.log" 。 但如果想要解析其他的日志文件，可以通过设置 session 变量 `tidb_slow_query_file` 为具体的文件路径，然后查询 `INFORMATION_SCHEMA.SLOW_QUERY` 就会按照设置的路径去解析慢日志文件。
```

File: sql/slow-query.md [hunk 18]
```diff
@@ -5,85 +5,208 @@ category: user guide
...
+Query OK, 0 rows affected
+Time: 0.001s
+```
 
-表示 SQL 语句
+`INFORMATION_SCHEMA.SLOW_QUERY` 目前暂时只支持解析一个慢日志文件，如果慢日志文件超过一定大小后被 logrotate 成多个文件了，查询 `INFORMATION_SCHEMA.SLOW_QUERY` 也只会去解析一个文件。后续我们会改善这点。
+
+####  用 pt-query-digest 工具分析 TiDB 慢日志
+
+可以用 pt-query-digest  工具分析 TiDB 慢日志，下面是示例。
```
Comment (bb7133) [inline sql/slow-query.md]:
```suggestion
可以用 pt-query-digest 工具分析 TiDB 慢日志，下面是示例。
```

**Comment by lilin90** [comment]
@IzabelWang Please help address the comments of @bb7133 in a new PR and help update the corresponding English document accordingly. Thanks!

File: sql/slow-query.md (hunk 14 above)
Comment (superlzs0476) [inline sql/slow-query.md]:
```suggestion
`Num_cop_tasks`：表示 [cop-tasks](https://github.com/pingcap/docs-cn/blob/master/sql/understanding-the-query-execution-plan.md) 的数目。
```

**Comment by crazycs520** [comment]
@lilin90 Already address comment in new PR: https://github.com/pingcap/docs-cn/pull/1266 and https://github.com/pingcap/docs-cn/pull/1271

**Comment by lilin90** [comment]
@crazycs520 I see that. Thanks! But there are still some other format issues. @IzabelWang helped fix them in #1277.

**Pull request merged**